specs/binding-kafka.spec/src/main/scripts/io/aklivity/zilla/specs/binding/kafka/schema/kafka.schema.patch.json

```diff
@@ -78,6 +78,172 @@
                                         {
                                             "type": "string",
                                             "enum": [ "none", "json_patch" ]
+                                        },
+                                        "key":
+                                        {
+                                            "type": "object",
+                                            "additionalProperties": false,
+                                            "properties":
+                                            {
+                                                "type":
+                                                {
+                                                    "title": "Type",
+                                                    "type": "string",
+                                                    "enum":
+                                                    [
+                                                        "byte_array",
+                                                        "byte_buffer",
+                                                        "bytes",
+                                                        "double",
+                                                        "float",
+                                                        "integer",
+                                                        "avro",
+                                                        "json",
+                                                        "json_decoder",
+                                                        "json_schema",
+                                                        "list",
+                                                        "long",
+                                                        "short",
+                                                        "string",
+                                                        "uuid",
+                                                        "void",
+                                                        "wrapper_key"
```

Comment:
Some of these type only seem to be relevant for cases where application code would observe the decoded value via Java API for example. However, we are implementing this inside `zilla` so `byte_array` would seem indistinguishable from `bytes`, i.e. just binary data.

Please remove any of these types that do not have distinct meaning for the encoded representation.